engine/runtime/src/main/java/org/enso/interpreter/node/callable/resolver/HostMethodCallNode.java

```diff
@@ -80,7 +80,9 @@ public boolean isInteropLibrary() {
    */
   public static PolyglotCallType getPolyglotCallType(
       Object self, String methodName, InteropLibrary library) {
-    if (library.isMemberInvocable(self, methodName)) {
+    if (library.isDate(self) && !library.isTime(self)) {
```

Comment:
Consider doing the same for `isString` – this introduces an inconsistency, where we can call polyglot methods on strings, but not on dates. It's fine for it to work this way, but should be consistent across other types. Long story short: move the `isString` case up :P

Reply:
I moved the check in [9611cb5](https://github.com/enso-org/enso/pull/3559/commits/9611cb55fceadcbcf7ed11ff63b7c013395ddabe), but it doesn't have any impact right now. I tried for example:
```
import Standard.Base.IO


main =
    h = hello
    IO.println h
    IO.println <| h.format "a{}b"

foreign python hello = """
    return '{:*^30}'.format('Hello world!')
```
and it fails with `Execution finished with an error: Method `format` of a polyglot object could not be found.` - the `h` in `h.format` is just `java.lang.String` and it has no polyglot properties.